### ISSUE TITLE
[proposal] Move to Material 3

### ISSUE THREAD

**Issue by paaspaas00** [futo-org/grayjay-android#1018]
I love this app! You guys are amazing!!

However, I noticed that the visual design is a bit weird. It is vaguely inspired by Google Material Design (1|2|3) but since some stuff don't really follow its guidelines, in the end the UX feels a bit strange.
Some examples:
- The bottom bar is too crammed, too many icons. Should just have one. The icons there don't have a clear feedback on whether they've been tapped on (at the moment, they just get some white accent color);
- the "more" menu in the bottom bar behaves like it was a menu from a FAB, but it isn't, which is weird to the user. The in/out transitions are non-standard and are nowhere to be found in other android apps.
- the settings menu opening animation is non standard and strange, an I find it very annoying because it doesn't have a fade in effect and feels clunky. See in immich for instance, the settings/backup in-out animations are much more pleasing there in my opinion.
- the settings list looks too crammed and difficult to follow and parse at a glance, especially because the text in the toggle options description is too light. Combobox buttons are not clear, with the background the same color as the window background, the text on them looks floating and detached to the down arrow symbol.
- buttons in general are too crammed, for instance those in the creators list, easy to misclick.

So in general, the UI is a bit suboptimal in my opinion.
I think Google's Material 3 apps are way better in this regard, and have a proper usability and user pref study behind, as well as clear guidelines and an easy to use UI framework already made, instead of rolling our own design paradigm.

I think one of the reasons why for instance people love Immich the app is the fact they embrace Material 3 (which is a consequence of the fact they want to copy the Google Foto UI), and this has the advantage of making the switch easier to users and the great advantage of adopting a UX with a very nice usability study already made by Google (which has UX experts, visual designers etc to guide its development).

So why not simply adopting Material 3, taking the YT app (for the player) as a source of inspiration and Immich for getting inspired about the settings? The general list of layout of the components is generally good in my opinion, it would just needs to be revamped with some visual adjustments from the Material 3 guidelines (there are many examples there that may fit our lists and menus for instance).

I'm sure people would love it and it would make the transition much more seamless for the average Joe coming from the Google ecosystem

**Comment by b-risk** [comment]
If it were to be implemented a new UI, a way to allow it to be themed similar to the plugin system would be awesome.

**Comment by paaspaas00** [comment]
Here is a nice video player app following Materia 3
https://github.com/anilbeesetti/nextplayer

The list view is very very clean and legible, I think it could be used as a source of inspiration for basically all the lists in GrayJay (videos, creators, downloads etc)
https://github.com/anilbeesetti/nextplayer/blob/main/fastlane/metadata/android/en-US/images/phoneScreenshots/3.png

**Comment by paaspaas00** [comment]
The transition should be pretty easy, see here https://developer.android.com/develop/ui/compose/designsystems/material2-material3

**Comment by Ubodduq** [comment]
@kaidelorenzo is this being considered? I'd absolutely love a Material 3 GUI!!

**Comment by kaidelorenzo** [comment]
I think there are definitely other priorities right now with regards to the desktop app. However on a personal note I would support a transition to a more material 3 like UI. I would say don't exactly get your hopes up but this kind of UI/UX discussion from users is very valuable considering we don't have any telemetry to inform decisions.

**Comment by cyraid** [comment]
@kaidelorenzo where is the desktop app? Is there a github?

**Comment by kaidelorenzo** [comment]
@cyraid are you wondering about a GitHub repo mirror?

**Comment by cyraid** [comment]
I was looking for the desktop link. If you go to https://grayjap.app/ and click on Download, there's just Android versions to download. For Desktop, just says "A desktop version is actively in the works, and already in internal testing phases.". Is this on purpose? Or just hasn't been updated?

**Comment by defiling9046** [comment]
Is there any update on this as I feel like this may be more important since I tried getting a friend to try Grayjay out and he tried it out but went back to Newpipe because the UI was easier to use/understand. 

So is there an ETA on this or do you want to implement some other features first and then this?

I also would love a refreshed UI because compared to the other apps I use like Aegis, Proton Mail and Immich Grayjay feels quite out of place as if the UI was a 2nd thought when I don't think it should be as I feel like more people would use the app (like my friend) if the UI had a higher priority.

Just my opinion tho.

P.S I just realised it's been over a year since this issue was first brought up now.

**Comment by cyraid** [comment]
@defiling9046 yeah I hear ya. With all the clutter of buttons under every video instead of a normal 3 dot action button we're all used to, and the messy popup when you hit more at the bottom, and the need for a header when it should be just some icons of search at the bottom to make it simple. Just so much distracting clutter. But I still use it.

**Comment by defiling9046** [comment]
> [@defiling9046](https://github.com/defiling9046) yeah I hear ya. With all the clutter of buttons under every video instead of a normal 3 dot action button we're all used to, and the messy popup when you hit more at the bottom, and the need for a header when it should be just some icons of search at the bottom to make it simple. Just so much distracting clutter. But I still use it.

100% I love the app don't get me wrong.

The concept with combining all the platforms together as I mainly use Grayjay for watching videos now is just a brilliant idea.

I think I remember them once saying about considering supporting RSS/articles etc and since the only "social media" app I have installed on my phone is currently Grayjay, if they added support for RSS and Twitter that would make Grayjay one of the best apps in my opinion.

@Zvonimir-FUTO said "There is a plan for supporting articles/blog posts content better but that will some time come after Shorts probably. People already opened FR for those platforms. Not sure how this will look and we are not sure do we want to add that type of content as well but that's gonna be decided in the far future."

So all of that combined with Material 3 would be a chefs kiss and I truly hope the app succeeds.

**Comment by defiling9046** [comment]
I thought I should add this here as it links in neatly with this. As someone who uses the Steam Deck, they released a UI update a slight while ago that I think Grayjay could definitely be inspired by (mostly in the settings page). Here's what they did down below:

Before they updated the UI:

<img width="2560" height="1440" alt="Image" src="https://github.com/user-attachments/assets/dc1568d8-a17a-45ca-b799-f19a4efa5a8c" />

After:

<img width="2560" height="1440" alt="Image" src="https://github.com/user-attachments/assets/fd1a0759-d3b7-44fd-a9df-ae4f0891d4cc" />

With their changelog for the update being:

**"Updated the layout and styles for client settings across desktop, Big Picture and controller configuration to improve readability and organization."**

I think Grayjay's settings could really do with something this as when I go through the settings at the moment it doesn't feel like a great experience @Zvonimir-FUTO

**Comment by defiling9046** [comment]
Someone made a rewrite of Grayjay into Material 3 for anyone curious!

It can probably do some things better but overall it's surprisingly good after trying it out myself.

Missing a few features that the main Grayjay app has like the hold to increase playback speed but seems like it has a good foundation none the less.

[redacted]

@Zvonimir-FUTO didn't know whether you'd be interested seeing this or not👍

Edit: Just realised this was created by the same guy who made the feature request😅

**Comment by Zvonimir-FUTO** [comment]
@defiling9046 could you please refrain from linking to external repositories containing unvetted code? We do not have the capacity to audit these sources for security or stability, so we cannot confirm they are safe for our users.

Hope you can understand. The PR will be reviewed separately.

**Comment by paaspaas00** [comment]
Hey folks, I'm the author of the material 3 rewrite above. Some time ago I contacted FUTO explaining I had this app almost done and asking them if they were interested (my repo was originally private, I offered Grayjay devs access if they wanted). My goal was to donate my rewrite to Grayjay devs so they can go on from there. My rewrite is completely (100%) done with AI (GPT 5.6 Sol Very High, about 10 days of work total, heavily directed and assisted for glitch and bug fixes).

Unfortunately nobody replied to my email, so I just made the thing public so people could try it. Hope you all try it out and if you appreciate you can fork it and continue Grayjay development using my codebase or portions of it. @Zvonimir-FUTO let me know what you devs think about it, it has been pretty fun and easy to develop I must say

**Comment by Zvonimir-FUTO** [comment]
Hey @paaspaas00 we are happy to see your enthusiasm for the Grayjay project and it's always fun to see the fan made forks. However, for now we cannot adopt those changes because of the sheer scope of what they affect and a lot of the affected portions had very deliberate decisions made. But we do appreciate the contribution regardless.

**Comment by cyraid** [comment]
@Zvonimir-FUTO Maybe if you guys are focusing on that at some time, you could minimize the button clutter on every video into a single 3 dot action popup? Make the feed videos a bit cleaner. :)
